### PR TITLE
Remove unnecessary recommendations from extensions.json

### PR DESCRIPTION
**Pull request by samuelFain**
#### Remove unnecessary extension recommendations from extensions.json
List of removed extensions:

-  "ms-python.black-formatter"
- "rangav.vscode-thunder-client"
- "Gruntfuggly.todo-tree"
- "shd101wyy.markdown-preview-enhanced"
- "timonwong.shellcheck"

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -62,14 +62,10 @@
 				"njpwerner.autodocstring",
 				"VisualStudioExptTeam.vscodeintellicode",
 				"yzhang.markdown-all-in-one",
```
Comment (ilaner) [inline .devcontainer/devcontainer.json:64]: do we need this?

**Comment by ilaner** [comment]
let's add to the list:
* ms-python.debugpy (better python debug experience)
* EditorConfig.EditorConfig (support the `.editorconfig` file we now have in content)
* DavidAnson.vscode-markdownlint (we already use it in sdk format, this lints and autofixes MD files.
* ms-vscode.powershell (powershell integrations...)

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (samuelFain) [reply .devcontainer/devcontainer.json:64]: Not really, I removed on extension (there were two) but the OOTB VS Code markdown support is good enough in my opinion.

**Review by ilaner — APPROVED** [review]
Great job

**Pull request merged**